dvclive: Add plots (#3142)
iterative/dvc.org: content/docs/dvclive/api-reference/sklearn.md

# Scikit-learn utilities

See [detailed usage examples](/doc/dvclive/ml-frameworks/sklearn).

## sklearn.log_confusion_matrix

Generates a [_Confusion Matrix_](https://en.wikipedia.org/wiki/Confusion_matrix)

> plot (usable by `dvc plots`) .

Reviewer: 💅 If you're already using `_` a lot in the DVCLive ref then it's OK but probably no need for italics in linked text that already stands out due to underscore and blue color. If anything it should be a CSS style

Reviewer: > Usable by dvc plots how?

Nvmd I see it's explained in the description. I'd just maybe put it like this instead:

```suggestion
Generates a [confusion matrix](https://en.wikipedia.org/wiki/Confusion_matrix)
(which can be visualized as `dvc plots`) .
```

> plot (usable by `dvc plots`) .

Reviewer: Usable by `dvc plots` how? Rhetorical: please clarify in the doc :)

```py
def log_confusion_matrix(y_true, y_pred, output_file):
```

#### Usage:

```py
from dvclive.sklearn import log_confusion_matrix

y_true = [1, 1, 2, 2]
y_pred = [2, 1, 1, 2]

log_confusion_matrix(y_true, y_score, "cm.json")
```

### Description

> Stores the predictions and labels in the format expected by the DVC
> [confusion matrix template](/doc/command-reference/plots#example-confusion-matrix).

Reviewer: ```suggestion
Stores the predictions and labels in the format expected by the confusion
matrix
[template](/doc/command-reference/plots#plot-templates-data-series-only)
of `dvc plots`.
```

Reviewer: Or maybe use your description from the other page. Seems more straightforward:

> set of... utilities that allow you to generate `dvc plots` from the labels and predictions...

### Parameters

> - `y_true` - Ground truth (correct) target values.
> 
> - `y_score` - Estimated targets as returned by a classifier..
> 
> - `output_file` - The name of the file where the `.json` plot will be saved.

Reviewer: 💅 The capitalization can work either way but in DVC docs we use lower case in these (up to you). The periods are not needed here though (incomplete sentences).
```suggestion
- `y_true` - ground truth (correct) target values

- `y_score` - estimated targets as returned by a classifier

- `output_file` - name of the file where the JSON plot will be saved
```

> - `output_file` - The name of the file where the `.json` plot will be saved.

Reviewer: First mention that the output is JSON. Should be stated earlier, I think.

# Scikit-learn utilities

> See [detailed usage examples](/doc/dvclive/ml-frameworks/sklearn).

Reviewer: Maybe
```suggestion
> See also a [detailed usage examples](/doc/dvclive/ml-frameworks/sklearn).
```
